### ISSUE TITLE
Engine hangs for half a second when selecting a node in 2D viewport or committing node rename.

### ISSUE THREAD

**Issue by JackSainsbury** [godotengine/godot#81665]
### Godot version

4.1.1.stable [bd6af8e0e]

### System information

Windows 10, Intel(R) Core(TM) i7-9750H, NVIDIA GeForce RTX 2070 Driver version 537.34 

### Issue description

https://github.com/godotengine/godot/assets/25502005/1ddebe15-17e7-48b5-a334-2afa506004d8

Selecting UI element in the 2D viewport and renaming a node in the inspector causes about a half second hang of the engine. Video recording was on a completely fresh project. All other aspects of the engine (that I have checked so far) seem to be working as expected - viewport navigation, menus, editing text to display on the button, seamless. 

Interestingly, while drag selecting, if you drag over only the button, the lag persists, if you drag over the button AND the scene in the viewport, they are instantly both selected in the hierarchy and view.

I have tried running as administrator, moving the exe and project to my ssd, external drive and internal HDD. Mobile and Fwd+ renderers tested, I have added the exe as a high priority GPU application in my display settings, I've tried running from cmd with '--disable-gpu', no errors or warnings being reported on the console version of the application.

Worth noting the NVidia driver I am running, as listed was installed today and is pretty bleeding edge.

### Steps to reproduce

Create new project, Mobile or Forward+ renderer, add 2D scene, add button, click button or drag select ONLY button.

Create new project, Mobile or Forward+ renderer, add 2D scene, rename 2D scene in Scene 'inspector' node panel.

### Minimal reproduction project

N/A

**Comment by Calinou** [comment]
>  I've tried running from cmd with '--disable-gpu', no errors or warnings being reported on the console version of the application.

[There is no `--disable-gpu` command line argument in Godot.](https://docs.godotengine.org/en/stable/tutorials/editor/command_line_tutorial.html)

**Comment by JackSainsbury** [comment]
Yep, I can confirm the issue persists in compatibility rendering.

**Comment by JackSainsbury** [comment]
@Calinou The renaming delay appears to be fixed in 4.2d4 however the selection delay in the 2d view persists as before.
